packages/colorpickers/src/elements/ColorSwatch/index.spec.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { createRef } from 'react';
+import { render, screen } from 'garden-test-utils';
+import userEvent from '@testing-library/user-event';
+import { ColorSwatch } from './index';
+
+const colors = [
+  [
+    { label: 'Green-800', value: '#0b3b29' },
+    { label: 'Green-700', value: '#186146' }
+  ],
+  [
+    { label: 'Green-600', value: '#038153' },
+    { label: 'Green-500', value: '#228f67' }
+  ]
+];
+
+describe('ColorSwatch', () => {
+  it('passes ref to underlying DOM element', () => {
+    const ref = createRef<HTMLTableElement>();
+
+    render(<ColorSwatch colors={colors} ref={ref} />);
+
+    expect(ref.current).toBe(screen.getByRole('grid'));
+  });
+
+  it('renders checkmark svg when a color is selected', () => {
+    render(<ColorSwatch colors={colors} />);
+
+    userEvent.tab();
+    expect(screen.getByTestId('#0b3b29').firstChild?.nodeName).not.toBeDefined();
```

Comment:
Should each color be labelled, that could then be used to query a specific button using `getByLabelText`? 

It would also help screen reader users. Currently it looks like sighted users see a tooltip with the label (for example "Green-700"), but the screen reader announces 'blank'. <img width="955" alt="Screen Shot 2021-09-17 at 11 30 05 AM" src="https://user-images.githubusercontent.com/88336617/133816150-1d603d6f-6224-4fa7-8c0f-ab95fea27d83.png">

Reply:
Hmm, that's strange! I ran it across two machines and the color is read aloud on the screen reader. Anyone else experiencing this issue?

<img width="500" alt="Screen Shot 2021-09-17 at 11 25 26 AM" src="https://user-images.githubusercontent.com/1811365/133836324-9a8e15e1-bb19-4f31-8ce3-f02bc513bc41.png">

<img width="500" alt="Screen Shot 2021-09-17 at 10 32 31 AM" src="https://user-images.githubusercontent.com/1811365/133834221-0d5fe2c7-c8b2-4818-9536-ac08dc52149e.png">

Reply:
The intended screen reader announcement when navigating color swatch chips is _"Color-XXX, blank"_. The accessibility model leverages tooltips rather than labels or titles. Each button, has a `aria-describedby` which points to an element with `role='tooltip`. This is how the color is announced to the user.

As for the "blank" that's being announced, that's because the button content is empty. I don't think a `aria-label` or `title` would address that. I think it should be okay.

Check out Microsoft's `SwatchColorPicker` — the accessibility model is similar, except they chose to use `aria-label` and `title`. The screen reader announcement is similar to the color swatch in this PR.

Reply:
Please see https://github.com/zendeskgarden/react-components/pull/1208#discussion_r711304118. I think there are structural a11y tree problems here. These buttons should read out in much the same way as the buttons that occupy the example toolbars on the website.

Reply:
Resolving this thread and consolidating conversation to https://github.com/zendeskgarden/react-components/pull/1208#discussion_r711304118